pkg/backend/filestate/stack.go

```diff
@@ -32,39 +32,37 @@ import (
 	"github.com/pulumi/pulumi/sdk/v3/go/common/util/result"
 )
 
-// Stack is a local stack.  This simply adds some local-specific properties atop the standard backend stack interface.
-type Stack interface {
-	backend.Stack
-	Path() string // a path to the stack's checkpoint file on disk.
-}
-
 // localStack is a local stack descriptor.
 type localStack struct {
 	ref      *localBackendReference // the stack's reference (qualified name).
-	path     string                 // a path to the stack's checkpoint file on disk.
-	snapshot *deploy.Snapshot       // a snapshot representing the latest deployment state.
+	snapshot **deploy.Snapshot      // a snapshot representing the latest deployment state, allocated on first use.
```

Comment:
Given that `**` is rare, it may be worth clarifying here that `nil` is a valid value for `*deploy.Snapshot`, and we want to remember that.